pyiso/caiso.py

```diff
@@ -424,42 +430,56 @@ def set_dt_index(self, df, date, hours, end_of_hour=True):
             offset = 0
 
         # create list of combined datetimes
-        dts = [datetime.combine(date, time(hour=(h+offset))) for h in hours]
+        dts = [datetime.combine(date, time(hour=(int(h)+offset))) for h in hours]
 
         # set list as index
         df.index = dts
 
         # utcify
         df.index = self.utcify_index(df.index)
 
-        # return
         return df
 
     def _generation_historical(self):
         # set up storage
         parsed_data = []
 
         # collect data
-        this_date = self.options['start_at'].date()
-        while this_date <= self.options['end_at'].date():
+        request_date = self.options['start_at'].astimezone(self.ca_tz).date()
+        local_end_at = self.options['end_at'].astimezone(self.ca_tz).date()
+        while request_date <= local_end_at:
             # set up request
-            url_file = this_date.strftime('%Y%m%d_DailyRenewablesWatch.txt')
+            url_file = request_date.strftime('%Y%m%d_DailyRenewablesWatch.txt')
             url = self.base_url_gen + url_file
 
             # carry out request
             response = self.request(url)
             if not response:
-                this_date += timedelta(days=1)
+                request_date += timedelta(days=1)
                 continue
 
-            # process both halves of page
-            for header in [1, 27]:
-                df = self.parse_to_df(response.text,
-                                      nrows=24, header=header,
-                                      delimiter='\t+')
+            dst_error_text = 'The supplied DateTime represents an invalid time.  For example, when the clock is ' \
+                             'adjusted forward, any time in the period that is skipped is invalid.'
+            header_idx = 1
+            for part in [1, 2]:  # process both halves of page (i.e. two parts)
+                num_data_rows = 24
+
+                # The day transitioning to daylight saving time adds extra erroneous lines of text.
+                if part == 1 and dst_error_text in response.text:
+                    num_data_rows = 29
+
+                df = self.parse_to_df(response.text, nrows=num_data_rows, header=header_idx, delimiter='\t+')
+
+                # The day transitioning to daylight saving time has errors in part two of the file that need removal.
+                if part == 2:
+                    df = df[df.THERMAL.map(str) != '#VALUE!']
```

Comment:
Lines 461-475 handle the erroneous lines added on transition dates into daylight saving time.